net/nimble/host/src/ble_gap.c

```diff
@@ -2182,6 +2322,47 @@ ble_gap_adv_active(void)
     return ble_gap_slave.op == BLE_GAP_OP_S_ADV;
 }
 
+#if MYNEWT_VAL(BLE_EXT_ADV)
+int ble_gap_adv_set_tx_power(int8_t tx_power)
+{
+    ble_hs_lock();
+
+    if (ble_gap_adv_active()) {
+        ble_hs_unlock();
+        return BLE_HS_EBUSY;
+    }
+
+    ext_adv_tx_pwr = tx_power;
+
+    ble_hs_unlock();
+
+    return 0;
+}
+
+int ble_gap_adv_set_phys(uint8_t primary_phy, uint8_t secondary_phy)
+{
+    ble_hs_lock();
+
+    if (ble_gap_adv_active()) {
+        ble_hs_unlock();
+        return BLE_HS_EBUSY;
+    }
+
+    if (primary_phy > BLE_HCI_LE_PHY_CODED ||
+        secondary_phy > BLE_HCI_LE_PHY_CODED) {
```

Comment:
so perhaps we could also check if both are non-zero and primary is not 2M?
I would skip this in case this function calls HCI command which returns an error immediately, but in this case it just stores both values so can be a bit hard to guess that error in other call may be due to invalid values stored here.